File: benchmark/interactive_function/prisoners/prisoners.py
import random

def play_optimal(n):
    pardoned = 0
    in_drawer = list(range(100))
    for _ in range(n):
        random.shuffle(in_drawer)
        all_found = True
        for prisoner in range(100):
            reveal = prisoner
            found = False
            for _go in range(50):
                card = in_drawer[reveal]
                if card == prisoner:
                    found = True
                    break
                reveal = card
            if not found:
                all_found = False
                break
        if all_found:
            pardoned += 1
    return (pardoned / n) * 100

if __name__ == '__main__':
    n = 1000000
    print(f"Optimal play wins (Python): {play_optimal(n):.1f}%")

File: benchmark/interactive_function/prisoners/prisoners.js
'use strict';

function playOptimal(n) {
    let pardoned = 0;
    const inDrawer = Array.from({length: 100}, (_, i) => i);

    for (let r = 0; r < n; r++) {
        // Shuffle
        for (let i = 99; i > 0; i--) {
            const j = Math.floor(Math.random() * (i + 1));
            [inDrawer[i], inDrawer[j]] = [inDrawer[j], inDrawer[i]];
        }

        let allFound = true;
        for (let prisoner = 0; prisoner < 100; prisoner++) {
            let found = false;
            let reveal = prisoner;
            for (let go = 0; go < 50; go++) {
                const card = inDrawer[reveal];
                if (card === prisoner) {
                    found = true;
                    break;
                }
                reveal = card;
            }
            if (!found) {
                allFound = false;
                break;
            }
        }
        if (allFound) pardoned++;
    }
    return (pardoned / n) * 100;
}

const n = 1000000;
console.log(`Optimal play wins (Node.js): ${playOptimal(n).toFixed(1)}%`);